schema.graphql

```diff
@@ -26,27 +26,60 @@ type User @entity {
 }
 
 type TokenBalance @entity {
-    id: ID!
-    balance: BigInt!
-    user: User!
-    token: String!
+  id: ID!
+  balance: BigInt!
+  user: User!
+  token: String!
 }
 
 type UnipoolBalance @entity {
-    id: ID!
-    balance: BigInt!
-    user: User!
-    unipool: String!
-    rewards:  BigInt!
-    rewardPerTokenPaid:  BigInt!
+  id: ID!
+  balance: BigInt!
+  user: User!
+  unipool: String!
+  rewards: BigInt!
+  rewardPerTokenPaid: BigInt!
 }
 
-
 type Unipool @entity {
-    id: ID!
-    periodFinish: BigInt
-    totalSupply : BigInt!
-    rewardRate: BigInt!
-    lastUpdateTime: BigInt
-    rewardPerTokenStored: BigInt!
+  id: ID!
+  periodFinish: BigInt
+  totalSupply: BigInt!
+  rewardRate: BigInt!
+  lastUpdateTime: BigInt
+  rewardPerTokenStored: BigInt!
+}
+
+type TokenDistroBalance @entity {
+  id: ID!
+  allocatedTokens: BigInt!
+  allocationCount: BigInt!
+  claimed: BigInt!
+  givback: BigInt!
+  givbackLiquidPart: BigInt!
+}
+
+type TokenAllocation @entity {
+  id: ID!
```

Comment:
add field `tokenDistroAddress: String!` to this entity too, and fill it in `saveTokenAllocation`

Reply:
`event.address.toHex()` gets the address of contract emitting the event